packages/flutter/test/material/popup_menu_test.dart

```diff
@@ -2209,6 +2209,60 @@ void main() {
     await tester.pumpAndSettle();
     expect(find.text('foo'), findsOneWidget);
   });
+
+  testWidgets('The opened menu should follow if the button\'s position changed', (WidgetTester tester) async {
+    final GlobalKey buttonKey = GlobalKey();
+
+    Widget buildFrame(double width, double height) {
+      return MaterialApp(
+        home: Scaffold(
+          body:  SizedBox(
+            height: height,
+            width: width,
+            child: Center(
+              child: PopupMenuButton<int>(
+                child: SizedBox(
+                  key: buttonKey,
+                  height: 10.0,
+                  width: 10.0,
+                  child: const ColoredBox(
+                    color: Colors.pink,
+                  ),
+                ),
```

Comment:
Could this all just be a `Text('Open menu')` with a key?

Reply:
let's keep it as is, this is helpful for calculating the position information of the menu.